File: ios/RCTMGL/RNMBImageUtils.m
//
//  RNMBImageUtils.m
//  RCTMGL
//
//  Created by Nick Italiano on 1/18/18.
//  Copyright © 2018 Mapbox Inc. All rights reserved.
//

#import "RNMBImageUtils.h"

@implementation RNMBImageUtils

+ (NSString *)createTempFile:(UIImage *)image
{
    NSString *fileID = [[NSUUID UUID] UUIDString];
    NSString *pathComponent = [NSString stringWithFormat:@"Documents/rctmgl-snapshot-%@.%@", fileID, @"png"];
    NSString *filePath = [NSHomeDirectory() stringByAppendingPathComponent: pathComponent];
    
    NSData *data = UIImagePNGRepresentation(image);
    [data writeToFile:filePath atomically:YES];
    
    return filePath;
}

+ (NSString *)createBase64:(UIImage *)image
{
    NSData *data = UIImagePNGRepresentation(image);
    return [NSString stringWithFormat:@"%@%@",  @"data:image/png;base64,", [data base64EncodedStringWithOptions:NSDataBase64EncodingEndLineWithCarriageReturn]];
}

@end

File: ios/RNMBX/RNMBImageUtils.swift
class RNMBImageUtils {
  static func createTempFile(_ image: UIImage) -> URL {
    let fileID = UUID().uuidString
    let pathComponent = "Documents/RNMBX-snapshot-\(fileID).png"
    
    let filePath = URL(fileURLWithPath: NSHomeDirectory()).appendingPathComponent(pathComponent)
    
    let data = image.pngData()
    try! data?.write(to: filePath, options: [.atomic])
    return filePath
  }
  
  static func createBase64(_ image: UIImage) -> URL {
    let data = image.pngData()
    let b64string : String = data!.base64EncodedString(options: [.endLineWithCarriageReturn])
    let result = "data:image/png;base64,\(b64string)"
    return URL(string: result)!
  }
}

